fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadReportingServiceImpl.java

```diff
@@ -61,23 +71,19 @@
 import org.springframework.stereotype.Service;
 
 @Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
 public class ReadReportingServiceImpl implements ReadReportingService {
 
+    private static final String ORDER_BY_REGEX_PATTERN = "^[0-9]*$";
```

Comment:
I just realized this. Why do we want to sort only based on the number of the column? This can be very difficult to maintain. Imagine people start building on this and suddenly we change the SQL that we run and we switch up the column order. Functionally nothing will break without ordering but if you use ordering, it's not gonna sort on the expected columns.

Why don't we switch this into providing the column name an order by that?

If this is something you really want to keep (column number ordering) then please add test cases to cover every single column ordering, otherwise we'll regress in the future.

Reply:
Since, Stretchy report is a generic functionality and can serve multiple reports in a generic way, knowing the actual column name for each report will be very difficult. 
Even if the column order changes user can select the correct index and run the report to receive the report as per the requirement.
Column idx method is adopted to keep it generic and to work with the current framework with minimal invasive changes.